### PR TITLE
feat: add workflow bot typescript template

### PR DESCRIPTION
**Pull request by dooriya**
Workflow bot template contains the basic code skeleton for a restify server hosted bot, as well as
- a `helloWorld` command to initialize a sample workflow.
- a `doStuff` adaptive card action to invoke an action.

E2E TEST: N/A

### CODE REVIEW

File: templates/bot/ts/workflow/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+    "name": "command-response-bot",
```
Comment (swatDong) [inline templates/bot/ts/workflow/package.json:2]: workflow-bot?

Comment (dooriya) [reply templates/bot/ts/workflow/package.json:2]: Good catch, the description should be updated as well

File: templates/bot/ts/workflow/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        "@microsoft/teamsfx": "^1.0.0",
+        "botbuilder": "~4.16.0",
+        "restify": "^8.5.1"
+    },
+    "devDependencies": {
+        "@types/restify": "8.4.2",
+        "env-cmd": "^10.1.0",
+        "nodemon": "^2.0.7",
+        "shx": "^0.3.4",
+        "ts-node": "~9.1.1",
```
Comment (swatDong) [inline templates/bot/ts/workflow/package.json:31]: #5663 updates ts, ts-node versions, tsconfig. Keep the consistency with those changes.

**Review by swatDong — DISMISSED** [review]

**Review by yukun-dong — DISMISSED** [review]

File: templates/bot/ts/workflow/README.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+| Folder | Contents |
+| - | - |
+| `.fx` | Project level settings, configurations, and environment information |
+| `.vscode` | VSCode files for local debug |
+| `bot` | The source code for the command and response Teams application |
+| `templates` | Templates for the Teams application manifest and for provisioning Azure resources |
+
+The core command-response implementation is in `bot` folder.
+
+The following files provide the business logic for command and response bot. These files can be updated to fit your business logic requirements. The default implementation provides a starting point to help you get started.
```
Comment (qinezh) [inline templates/bot/ts/workflow/README.md:40]: > The following files provide the business logic for command and response bot.

update the README as well?

Comment (dooriya) [reply templates/bot/ts/workflow/README.md]: Updated. This is a draft version, and will work with @MuyangAmigo to consolidate the document.

**Review by qinezh — APPROVED** [review]

**Review by yukun-dong — APPROVED** [review]

File: templates/bot/ts/workflow/src/adaptiveCards/doStuffActionResponse.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    {
+      "type": "ActionSet",
+      "actions": [
+        {
+          "type": "Action.OpenUrl",
+          "title": "Bot Framework Docs",
+          "url": "https://docs.microsoft.com/en-us/azure/bot-service/?view=azure-bot-service-4.0"
+        },
+        {
+          "type": "Action.OpenUrl",
```
Comment (MuyangAmigo) [inline templates/bot/ts/workflow/src/adaptiveCards/doStuffActionResponse.json:24]: I feel like this default action is too simple and would make it difficult to extend.. I suggest we can have an action that responds with another card so it showcases "workflows" are triggered by actions and steps are chained by responding card.

File: templates/bot/ts/workflow/src/cardActions/helloworldActionHandler.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import { AdaptiveCards } from "@microsoft/adaptivecards-tools";
```
Comment (MuyangAmigo) [inline templates/bot/ts/workflow/src/cardActions/helloworldActionHandler.ts:1]: Can we have consistent naming on action handler and its corresponding response card? like `helloWorldActionHandler.ts` and `helloWorldActionResponse.json`

File: templates/bot/ts/workflow/src/internal/initialize.ts [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+// You can also use the commandBot.command.registerCommands to register other commands
+// if you don't want to register all of them in the constructor
+export const commandBot = new ConversationBot({
+  // The bot id and password to create BotFrameworkAdapter.
+  // See https://aka.ms/about-bot-adapter to learn more about adapters.
+  adapterConfig: {
+    appId: process.env.BOT_ID,
+    appPassword: process.env.BOT_PASSWORD,
+  },
+  command: {
```
Comment (MuyangAmigo) [inline templates/bot/ts/workflow/src/internal/initialize.ts:15]: Probably to add a comment here why command is needed here - to initiate the workflow?

File: templates/bot/ts/workflow/src/adaptiveCards/doStuffActionResponse.json (hunk 4 above)
Comment (dooriya) [reply templates/bot/ts/workflow/src/adaptiveCards/doStuffActionResponse.json:24]: If user want to extend the workflow, they can repeat the steps (1234) as showed in the `README`. For default template, I think just one action can demonstrate the code for card actions.

**Pull request merged**